### PR TITLE
Feature : add IoC dependency injection

### PR DESCRIPTION
**Pull request by heiso**
PR based on the great work of @patrickmichalina https://github.com/Boulangerie/hapiour-decorators/pull/10

### CODE REVIEW

**Comment by heiso** [comment]
Hey @patrickmichalina !
Think I'm done with this feature, feel free to test it :)

File: lib/core/bootstrap.ts [hunk 1]
```diff
@@ -46,16 +51,16 @@ export function bootstrap(...BootstrapedApps: Array<IAppStatic>): Array<IApp> {
   return apps
 }
 
-function getPluginsRecurs(Plugins: Array<IPluginStatic|IPlugin|IPluginConfiguratorStatic|Array<IPluginStatic|IPlugin|IPluginConfiguratorStatic>>): Array<IPlugin> {
+function getPluginsRecurs(injector: IInjector, Plugins: Array<IPluginStatic | IPlugin | IPluginConfiguratorStatic | Array<IPluginStatic | IPlugin | IPluginConfiguratorStatic>>): Array<IPlugin> {
```
Comment (heiso) [inline lib/core/bootstrap.ts]: check type of Plugins.

**Review by yannickglt — APPROVED** [review]
LGTM

**Comment by patrickmichalina** [comment]
LGTM, thanks

**Comment by heiso** [comment]
Something is wrong with this build, not working with one of my tricky project, fixin' an then, merge it :)

**Pull request merged**